build.rs

```diff
@@ -397,9 +397,14 @@ fn ends_with(entry: &DirEntry, pat: &str) -> bool {
 /// [1]: https://github.com/python/cpython/blob/3.5/Lib/sysconfig.py#L389
 fn find_sysconfigdata(cross: &CrossCompileConfig) -> Result<PathBuf> {
     let sysconfig_paths = search_lib_dir(&cross.lib_dir, &cross);
+    let sysconfig_name = env::var_os("_PYTHON_SYSCONFIGDATA_NAME");
     let mut sysconfig_paths = sysconfig_paths
         .iter()
-        .filter_map(|p| fs::canonicalize(p).ok())
+        .filter_map(|p| {
+            fs::canonicalize(p)
+                .ok()
+                .filter(|p| p.file_stem() == sysconfig_name.as_deref())
```

Comment:
I've changed `file_name()` to `file_stem()` since `_PYTHON_SYSCONFIGDATA_NAME` does not include the `.py` extension.